modules/porous_flow/src/userobjects/AdvectiveFluxCalculatorBase.C

```diff
@@ -374,67 +492,104 @@ AdvectiveFluxCalculatorBase::finalize()
     for (const auto & neighbors : nodes.second)
     {
       const dof_id_type j = neighbors.first;
-      const Real u_j = getU(j);
+      const Real u_j = getUnodal(j);
 
       // negative sign because residual = -Lu (KT equation (19))
       _flux_out[i] -= lij[i][j] * u_j + fa[i][j];
 
       _dflux_out_du[i][j] -= lij[i][j];
       for (const auto & dof_deriv : dfa[i][j])
         _dflux_out_du[i][dof_deriv.first] -= dof_deriv.second;
+
+      _dflux_out_dKjk[i][i][j] -= 1.0 * u_j; // from the K in L = K + D
+      /* NOTE: put back in and change _kij[i][j] to lij[i][j] in 2 places above, andremove the 0 *
```

Comment:
andremove -> and remove

Reply:
ooops, that line should be removed entirely